adagrams/game.py

```diff
@@ -1,11 +1,135 @@
+    # a list to store all letters from a pool of letter
+    list_letter = []
+    # loop over LETTER_POOL to get key and value
+    # key will be the letter adding to list and 
+    # value is number of times that the letter will be adding
+    for letter, count in LETTER_POOL.items():
+        # adding letter into list depend on value of count 
+        for i in range(count):
+            list_letter.append(letter)
+    # create a list that randomly draw 10 letters from list_letter 
+    letter_draw = random.sample(list_letter, k=10)
+    
+    return letter_draw
 
+# Wave 2
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    Checking each letter of word:
+        - return False if it is not in letter_bank
+        - return True if it is in letter_banks
+    """
+    # create a new list which same as letter_bank, so it doen't effect to original when editing
+    letter_bank_copy = letter_bank.copy()
+    # loop over word to get each character
+    for char in word.upper():
+        # return False if the character of word not in the new created list
+        if char not in letter_bank_copy:
+            return False
+        # otherwise removing each character of word from letter_bank_copy when it is in there
+        # return True after removing all characters of word that is in letter_bank_copy
+        else: 
+            letter_bank_copy.remove(char)
+    return True
 
+# Wave 3
 def score_word(word):
-    pass
+    """
+    Sum score of each character of input
+        - return 0 if input is empty, integer, float, or characters of number.
+        - otherwise,sum point if the character of input is uppercase or lowercase and it is in table of point value
+        - if input has special character then ignore that special character
+        - return total score at the end
+    """
+    # create a variable to sum score for all valid input
+    total_score = 0
+    # return 0 if input is empty or not string
+    if (len(word) == 0) or (type(word) != str) or word.isnumeric():
+        return 0
+
+    # loop over char_point dictionary to get key and value for comparing
+    for letter, point in LETTER_POINT.items():
+        # loop over the word to get each character and check whether is it in the key of char_point
+        for char in word:
+            # get point value from char_point dictionary to sum if character of word is string and it is in tuple key of dictionary
+            if char.upper() in letter:
+                total_score += point
 
+    # if length of word is from 7 to 10, add additional 8 point
+    if len(word) > 6 and len(word) < 11:
+        total_score += 8
+    return total_score
+
+# Wave 4
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """Finding higest score and return the word and score in Tuple."""
+
+    # find max score
+    max_score = max([score_word(word) for word in word_list])
```

Comment:
Nice comprehension! As a minor simplification, we can omit the square brackets when immediately passing a comprehension into a function that takes an iterable:
```
max_score = max(score_word(word) for word in word_list)
```